### PR TITLE
expand rp tests

### PR DESCRIPTION
**Pull request by andre-merzky**
This PR expands the set of tests for RP - they now include a smoke test, unit tests and component tests.

The PR also contains some (commented out) code to run multiple tests in the same GH workflow, and to fail the test if any of them fails.

### CODE REVIEW

**Comment by andre-merzky** [comment]
This PR needs work - RP unit tests should have failed (and did), but GH Actions show success.  Will look into it.

**Comment by SteVwonder** [comment]
Good catch!  It looks like [github CI's `step.run` uses fail-fast](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsrun) (e.g., `bash -e`) so if any of the `docker run`s fail, that step should fail.  It also appears that [`docker run` propagates the exit code of the command up](https://docs.docker.com/engine/reference/commandline/run/#assign-name-and-allocate-pseudo-tty---name--it).  I wonder if the issue is that the `run_rp_test.sh` script does not have `set -e`, so the last line succeeds and the exit code is 0 despite an earlier command failing?

**Comment by andre-merzky** [comment]
This should be ready for review now.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,6 +23,22 @@ jobs:
     - name: Run tests
       run: |
         docker run rp
+        ret_rp=$?
+        echo "rp   : $ret_rp"
+        # docker run flux
```
Comment (SteVwonder) [inline .github/workflows/ci.yml:28]: Looks like the `docker run`s are commented out?  Maybe a leftover from debugging?

**Review by SteVwonder — CHANGES_REQUESTED** [review]
Thanks @andre-merzky.  Looks like the docker runs are commented out still:

File: .github/workflows/ci.yml (hunk 1 above)
Comment (andre-merzky) [reply .github/workflows/ci.yml:28]: That's on purpose actually - the PR was only supposed to expand the RP tests and to document how multiple tests can be run, and I did not want to expand scope by adding those tests.  But, OTOH, if you think it makes sense I'd be happy to enable them - makes the PR cleaner also...  What do you think?

Comment (SteVwonder) [reply .github/workflows/ci.yml:28]: Yeah, I'd vote for adding the `ret_X` around the various docker runs, but adding them around the `rp_parsl`, `rp_parsl_swift`, and `rp_parsl_swift_flux` images run below (it is the last component in the name whose CMD is run, so running `rp_parsl_swift` executes the swift tests, rp and parsl just happen to also be installed in the image.

Comment (SteVwonder) [reply .github/workflows/ci.yml:28]: PS - Assuming that doesn't require any fixups in the other components' images/tests.  If it does, then I think this PR is fine as is.

**Comment by dongahn** [comment]
@andre-merzky or @SteVwonder: is this ready to go in?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (andre-merzky) [reply .github/workflows/ci.yml]: Enabling the tests do result in failures, specifically on `docker run flux`:
```
Unable to find image 'flux:latest' locally
docker: Error response from daemon: pull access denied for flux, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
Error: Process completed with exit code 125.
```
I don't mind using this PR to fix this, or to leave as is.  @SteVwonder : what would you prefer?

Comment (SteVwonder) [reply .github/workflows/ci.yml]: We can leave as is.  Probably better to get this in sooner rather than later. Thanks @andre-merzky!

**Comment by andre-merzky** [comment]
Tests should pass now - if so this would be ready to merge.

**Review by SteVwonder — APPROVED** [review]
LGTM! Thanks @andre-merzky

**Comment by SteVwonder** [comment]
Ah. Github workflow was modified so mergify can't merge.  Doing so manually.

**Pull request merged**